pfcpiface/parse-pdr.go

```diff
@@ -1,28 +1,224 @@
 // SPDX-License-Identifier: Apache-2.0
 // Copyright 2020 Intel Corporation
+// Copyright 2022 Open Networking Foundation
 
 package main
 
 import (
 	"errors"
 	"fmt"
+	"math"
 	"net"
 
 	log "github.com/sirupsen/logrus"
 	"github.com/wmnsk/go-pfcp/ie"
 )
 
+// portFilter encapsulates a L4 port range as seen in SDF.
+type portFilter struct {
+	portLow  uint16
+	portHigh uint16
+}
```

Comment:
I think there is some value in encapsulating the port filter as a data type with exposed high-level functions.